pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -4552,6 +4555,75 @@ private CompletableFuture<Void> createSubscriptions(TopicName topicName, int num
         return result;
     }
 
+    /**
+     * It creates subscriptions for new partitions of existing partitioned-topics.
+     *
+     * @param topicName     : topic-name: persistent://prop/cluster/ns/topic
+     * @param numPartitions : number partitions for the topics
+     *
+     */
+    private CompletableFuture<Void> createMissedSubscriptionsAsync(TopicName topicName, int numPartitions) {
+        CompletableFuture<Void> result = new CompletableFuture<>();
+        PulsarAdmin admin;
+        try {
+            admin = pulsar().getAdminClient();
+        } catch (PulsarServerException e1) {
+            result.completeExceptionally(e1);
+            return result;
+        }
+        admin.topics().getStatsAsync(topicName.getPartition(0).toString()).thenAccept(stats -> {
```

Comment:
The partition-0 maybe also created by the `CreateMissedPartitions`, I think we should list the subscriptions of the partitioned topic.

And for Pulsar, users can subscription to a specific partition, which means partition-0, partition-1 might have different subscriptions.

Reply:
Can I understand this: get the TopicStats of all partitions of the Topic, and extract the public subscriptions of all TopicStats. If the TopicStats of the partition exist, do not execute createSubscriptionAsync. If the TopicStats of the partition do not exist, execute createSubscriptionAsync to create public subscriptions for the partition.

> The partition-0 maybe also created by the `CreateMissedPartitions`, I think we should list the subscriptions of the partitioned topic.



> And for Pulsar, users can subscription to a specific partition, which means partition-0, partition-1 might have different subscriptions.

Reply:
> The partition-0 maybe also created by the `CreateMissedPartitions`, I think we should list the subscriptions of the partitioned topic.
> 
> And for Pulsar, users can subscription to a specific partition, which means partition-0, partition-1 might have different subscriptions.

I think we can add a parameter to specify the subscription name list to be created, which can solve these two problems at the same time. How about this？

Reply:
@Pomelongan For the second one, it's not a problem. The partitions can have different subscriptions. The challenge is that we don't have subscriptions for the partitioned topic, right?

Hmmm, IMO, we can update the documentation to remind the user that "If you found the partitions is missed, you should use create missed partitions command to fix the missed partition issue and try to subscribe to the partitioned topic again to repair the part of the subscription missed issue"

Reply:
> The challenge is that we don't have subscriptions for the partitioned topic, right?

 @codelipenghui Yes, adding a parameter to specify the subscription list to be created can solve this problem. I have updated the code,PTAL

> Hmmm, IMO, we can update the documentation to remind the user that "If you found the partitions is missed, you should use create missed partitions command to fix the missed partition issue and try to subscribe to the partitioned topic again to repair the part of the subscription missed issue"

Okay, I have updated the document.